### ISSUE TITLE
[Bug]: Project export fails (only locally?)

### ISSUE THREAD

**Issue by kleintom** [SpeciesFileGroup/taxonworks#3966]
### Steps to reproduce the bug

```bash
1. Signin as (project) admin
2. Go to a project you're admin for and select Download Project Task
3. Click on Generate for Export SQL
4. Run the background job

Get a failure on `no function flatten! for nil` (or similar).

(This worked for a project on sandbox - maybe port isn't provided in the config there?)
```

### Environment

Development (docker)

**Comment by kleintom** [comment]
When I locally imported the project I exported from sandbox, my sandbox password didn't seem to work locally (which was fine, I just created a new admin account and then changed it) - is that expected? I tried to type the password carefully several times, and the user exists.

**Comment by kleintom** [comment]
Also on the imported project my user General Preferences were blanked, which was causing a Preferences.layout exception when I went to the Create CE task, for example. Resetting the preferences fixed it, but it took a minute to figure out what the problem was.

**Comment by LocoDelAssembly** [comment]
> When I locally imported the project I exported from sandbox, my sandbox password didn't seem to work locally (which was fine, I just created a new admin account and then changed it) - is that expected? I tried to type the password carefully several times, and the user exists.

Yes, this is normal, passwords are not transferred so no one can get a copy to perform offline brute force attacks on them. Also, users that do not belong to the project you are downloading get their email and name changed. Take a look: https://github.com/SpeciesFileGroup/taxonworks/blob/699af625a2b3fe969851a34e0ea9ad2b100724a0/lib/export/project_data/sql.rb#L139

**Comment by LocoDelAssembly** [comment]
> Also on the imported project my user General Preferences were blanked, which was causing a Preferences.layout exception when I went to the Create CE task, for example. Resetting the preferences fixed it, but it took a minute to figure out what the problem was.

Could this be a cookie problem?

 @mjy @jlpereira

**Comment by jlpereira** [comment]
> > Also on the imported project my user General Preferences were blanked, which was causing a Preferences.layout exception when I went to the Create CE task, for example. Resetting the preferences fixed it, but it took a minute to figure out what the problem was.
> 
> Could this be a cookie problem?
> 
> @mjy @jlpereira

Nope, user preferences are not saved in cookies

**Comment by kleintom** [comment]
Would it be helpful to add a note at the export button saying something like 'All passwords are reset to 'taxonworks' on export', or are most people going to be in a situation where someone will just tell them that?

> They [user preferences] are saved on users profile, maybe it's not being exported?

Doesn't look like it, I can work on that unless somebody else is already on it, let me know. Thanks!

**Comment by mjy** [comment]
>  add a note at the export button saying something like 'All passwords are reset to 'taxonworks' on export',

Sure.

> Doesn't look like it, I can work on that unless somebody else is already on it, let me know. Thanks!

Also yes.
